router/batchrouter/asyncdestinationmanager/snowpipestreaming/snowpipestreaming.go

```diff
@@ -149,10 +163,23 @@ func (m *Manager) Upload(asyncDest *common.AsyncDestinationStruct) common.AsyncU
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
+	// backoff should be reset if authz error is not encountered for any of the tables
+	shouldResetBackoff := true
 
 	discardsChannel, err := m.initializeChannelWithSchema(ctx, asyncDest.Destination.ID, &destConf, discardsTable(), discardsSchema())
 	if err != nil {
-		return m.abortJobs(asyncDest, fmt.Errorf("failed to prepare discards channel: %w", err).Error())
+		var authzErr *snowpipeAuthzError
+		if errors.As(err, &authzErr) {
+			// Ignoring this error so that the jobs are marked as failed and not aborted since
+			// we want these jobs to be retried the next time.
+			m.logger.Warnn("Failed to initialize channel with schema",
+				logger.NewStringField("table", discardsTable()),
+				obskit.Error(err),
+			)
+			shouldResetBackoff = false
```

Comment:
We can return from here itself. No point in going forward.
```suggestion
			m.stats.jobs.failed.Count(len(asyncDest.FailedJobIDs))

			return common.AsyncUploadOutput{
				FailedJobIDs:  asyncDest.FailedJobIDs,
				FailedCount:   len(asyncDest.FailedJobIDs),
				FailedReason:  fmt.Errorf("failed to initialize channel with schema: %w", err).Error(),
				DestinationID: asyncDest.Destination.ID,
			}
```

Reply:
Added an early return with failed job reporting